### DISCUSSION TITLE
Is SX1276SetAntSwLowPower() logic inverted?

### DISCUSSION THREAD

**Discussion by pieterconradie** [Lora-net/LoRaMac-node#1006] [Q&A]
Hi there,

In `SX1276SetOpMode()` is the following block:
```c
if( opMode == RF_OPMODE_SLEEP )
{
    SX1276SetAntSwLowPower( true );
}
else
{
    // Enable TCXO if operating mode different from SLEEP.
    SX1276SetBoardTcxo( true );
    SX1276SetAntSwLowPower( false );
    SX1276SetAntSw( opMode );
}
```
So it appears that one has to call `SX1276SetAntSwLowPower()` with `true` to select low power mode but the function content itself is confusing:
```c
/*!
 * Flag used to set the RF switch control pins in low power mode when the radio is not active.
 */
static bool RadioIsActive = false;

void SX1276SetAntSwLowPower( bool status )
{
    if( RadioIsActive != status )
    {
        RadioIsActive = status;

        if( status == false )
        {
            SX1276AntSwInit( );
        }
        else
        {
            SX1276AntSwDeInit( );
        }
    }
}
```

The variable name `RadioIsActive` seems to be logically inverted with regards to it's use?

Can this be fixed so that it makes sense or is it only me?

Thanks in advance,
Pieter

**Comment by mluis1** [comment]
When `status`parameter is `true` it means the radio is in sleep mode and we need to call `SX1276AntSwDeInit` and when `false` (all other operating modes) it means we need to call `SX1276AntSwInit`.   
The `RadioIsActive` variable is there to store the provided `status` parameter.  
The `RadioIsActive` is used to prevent the `SX1276AntSwInit` and `SX1276AntSwDeInit` function to be called every time the radio operating mode is not changed. For example if one calls `SX1276SetOpMode` with same `opMode` 2 or more times the `SX1276SetAntSwLowPower` is only executed once.

I can understand why it is confusing however I don't think there is a problem with `RadioIsActive`.

In the future it would be nice if you could post this kind of questions on the project [Discussions](https://github.com/Lora-net/LoRaMac-node/discussions) tab. It is a better place to engage discussions and then we can agree if it is an issue or not.
If you don't mind I move this issue to the [Discussions](https://github.com/Lora-net/LoRaMac-node/discussions) tab.
